From: Baolin Wang <baolin.wang@linux.alibaba.com>
Subject: mm: compaction: count the migration scanned pages events for proactive compaction
Date: Tue, 10 Jan 2023 21:36:20 +0800

The proactive compaction will reuse per-node kcompactd threads, so we
should also count the KCOMPACTD_MIGRATE_SCANNED and KCOMPACTD_FREE_SCANNED
events for proactive compaction.

Link: https://lkml.kernel.org/r/b7f1ece1adc17defa47e3667b5f9fd61f496517a.1673342761.git.baolin.wang@linux.alibaba.com
Signed-off-by: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: Matthew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/compaction.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/mm/compaction.c~mm-compaction-count-the-migration-scanned-pages-events-for-proactive-compaction
+++ a/mm/compaction.c
@@ -2660,6 +2660,11 @@ static void proactive_compact_node(pg_da
 		cc.zone = zone;
 
 		compact_zone(&cc, NULL);
+
+		count_compact_events(KCOMPACTD_MIGRATE_SCANNED,
+				     cc.total_migrate_scanned);
+		count_compact_events(KCOMPACTD_FREE_SCANNED,
+				     cc.total_free_scanned);
 	}
 }
 
_
